vmdb/app/models/host.rb

```diff
@@ -1393,16 +1393,25 @@ def self.get_hostname(ipAddress)
 
   def connect_ssh(options={})
     require 'MiqSshUtil'
-    if self.has_authentication_type?(:remote)
-      rl_user, rl_password = self.auth_user_pwd(:remote)
-      su_user, su_password = self.auth_user_pwd(:root)
-    else
-      rl_user, rl_password = self.auth_user_pwd(:root)
+    if self.kind_of?(HostOpenstackInfra)
+      # HostOpenstackInfra is using auth key set on ext_management_system level, not individual hosts
+      rl_user, auth_key = self.auth_user_keypair(:keypair)
+      options[:key_data] = auth_key
+      rl_password = nil
+      # TODO(lsmola) make sudo user work. So it with be optional sudo password for private key auth
       su_user, su_password = nil, nil
-    end
+    else
+      if self.has_authentication_type?(:remote)
+        rl_user, rl_password = self.auth_user_pwd(:remote)
+        su_user, su_password = self.auth_user_pwd(:root)
+      else
+        rl_user, rl_password = self.auth_user_pwd(:root)
+        su_user, su_password = nil, nil
+      end
```

Comment:
Or maybe this... either way, something so this code lives outside of connect_ssh.

```ruby
class Host
  def ssh_users_and_passwords
    if self.has_authentication_type?(:remote)
      [self.auth_user_pwd(:remote), self.auth_user_pwd(:root)]
    else
      [self.auth_user_pwd(:root), nil, nil]
    end
  end
end
```